10up/wp-vulnerability-scanner
# Add support for Patchstack API

### Description of the Change
As requested in #70, this PR adds support for Patchstack API to check reported vulnerabilities for WordPress core, installed plugins and themes.

Users can choose Patchstack API for vulnerabilities scan by adding `VULN_API_PROVIDER` constant in wp-config.php. By default, the current WPScan API service will be used.

```
define( 'VULN_API_PROVIDER', 'patchstack' );
```

Closes #70 

### How to test the Change
1. Add a `VULN_API_PROVIDER` constant in wp-config.php as below.
```
define( 'VULN_API_PROVIDER', 'patchstack' );
```
2. Add a `VULN_API_TOKEN` constant in wp-config.php by getting API key from patchstack.com
```
define( 'VULN_API_TOKEN', 'YOUR_TOKEN_HERE' );
```
3. Try given `vuln` commands in [readme](https://github.com/10up/wpcli-vulnerability-scanner#usage) to scan for vulnerabilities.
4. Make sure all commands are working fine.
4. Make sure all commands are working fine with both services (WPScan and Patchstack)

### Changelog Entry

> Added - Support for Patchstack API  

### Credits

Props @iamdharmesh, @MARQAS, @dkotter, @jeffpaul.

### Checklist:
